src/index.ts

```diff
@@ -1,17 +1,115 @@
+import {
+  getHubApiUrl,
+  getPortalApiUrl,
+  getSiteById,
+  IDomainEntry,
+  IHubRequestOptions,
+  lookupDomain,
+} from '@esri/hub-common';
+import { IContentSearchRequest } from '@esri/hub-search';
+import { Request, Response } from 'express';
+import * as _ from 'lodash';
 import { version } from '../package.json';
+import { getDataStreamDcatAp201 } from './dcat-ap';
+import * as config from 'config';
 
 export = class Output {
-    static type = 'output';
-    static version = version;
-    static routes = [
-      {
-        path: '/content',
-        methods: ['get'],
-        handler: 'serve'
+  static type = 'output';
+  static version = version;
+  static routes = [
+    {
+      path: '/dcat-ap/2.0.1',
+      methods: ['get'],
+      handler: 'serve',
+    },
+  ];
+
+  model: any;
+
+  public async serve (req: Request, res: Response) {
+    res.set('Content-Type', 'application/json');
+
+    try {
+      const { domainRecord, siteModel } = await this.fetchDomainAndSite(req.hostname);
+
+      const siteCatalog = _.get(siteModel, 'data.catalog');
+      if (!siteCatalog) {
+        res.status(200).send({});
+      } else {
+       const dcatStream = getDataStreamDcatAp201({
+        domainRecord,
+        siteItem: siteModel.item,
+        env: this.getEnvFromPortal(this.portalUrl),
+      });
+
+      req.res.locals.searchRequest = this.getSearchRequestFromCatalog(siteCatalog, this.portalUrl);
+
+      const datasetStream = await this.model.pullStream(req);
+
+      datasetStream
+        .pipe(dcatStream)
+        .pipe(res)
+        .on('error', (err: any) => {
+          res.status(500).send(this.getErrorResponse(err));
+        });
+      }
+    } catch (err) {
+      res.status(500).send(this.getErrorResponse(err));
+    }
+  }
+
+  private async fetchDomainAndSite (hostname) {
+    const requestOptions = this.getRequestOptions(this.portalUrl);
+
+    const domainRecord = (await lookupDomain(
+      hostname,
+      requestOptions,
+    )) as IDomainEntry;
+    const siteModel = await getSiteById(domainRecord.siteId, requestOptions);
+
+    return { domainRecord, siteModel };
+  }
+
+  private getRequestOptions (portalUrl: string): IHubRequestOptions {
+    return {
+      isPortal: false,
+      hubApiUrl: getHubApiUrl(portalUrl),
+      portal: getPortalApiUrl(portalUrl),
+      authentication: null,
+    };
+  }
+
+  private get portalUrl () {
+    return config.has('arcgisPortal')
+      ? config.get('arcgisPortal') as string
+      : 'https://www.arcgis.com';
+  }
```

Comment:
Is necessary to have this as a private `get`?  Why not just do this logic once at the top of the module:

```
const portalUrl = config.has('arcgisPortal')
      ? config.get('arcgisPortal') as string
      : 'https://www.arcgis.com';
```

This logic would only execute once when the module loaded as opposed to each time the `serve` method executed.